src/main/scala/rocket/PTW.scala

```diff
@@ -153,7 +153,7 @@ class PTW(n: Int)(implicit edge: TLEdgeOut, p: Parameters) extends CoreModule()(
     Cat(r_pte.ppn, vpn_idx) << log2Ceil(xLen/8)
   }
   val fragmented_superpage_ppn = {
-    val choices = (pgLevels-1 until 0 by -1).map(i => Cat(r_pte.ppn >> (pgLevelBits*i), r_req.addr(pgLevelBits*i-1, 0)))
+    val choices = (pgLevels-1 until 0 by -1).map(i => Cat(r_pte.ppn >> (pgLevelBits*i), r_req.addr(((pgLevelBits*i) min vpnBits)-1, 0).padTo(pgLevelBits*i)))
```

Comment:
How can (pgLevels - 1) * pgLevelBits ever exceed vpnBits?

Reply:
in a heterogeneous system where some cores have vmEnabled=true and some cores have vmEnabled=false, the vmEnabled=false cores still compile PTW.scala with a small number of vpnBits, leading to a Chisel compile error here before this all gets optimized away in Firrtl

Reply:
Right.  I guess I'm wondering why this wasn't a problem before... maybe all the systems had just enough PA bits so that by coincidence it wasn't a problem.